### PR TITLE
Jenkins

### PR DESCRIPTION
**Pull request by Awesome5040**
Jenkins

### CODE REVIEW

File: JenkinsFile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            steps {
+                sh './gradlew bootjar'
+            }
+        }
+        stage('Deploy jar to remote ec2') {
+            steps {
+                sshPublisher(
+                        continueOnError: false,
+                        failOnError: true,
+                        publishers: [sshPublisherDesc(configName: "stage", transfers: [sshTransfer(sourceFiles: 'job-search.jar')], verbose: true)])
```
Comment (oleksandr-fomenko) [inline JenkinsFile]: suggesting to extract sourceFiles name to the env variable

File: JenkinsFile [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            }
+        }
+
+        stage('Stop application') {
+            steps {
+                script {
+                    sshagent(credentials: ['ec2']) {
+                        sh 'ssh -o "StrictHostKeyChecking=no" -l ec2-user 172.31.88.196 "sudo fuser -k 8080/tcp || true"'
```
Comment (oleksandr-fomenko) [inline JenkinsFile]: suggesting to extract IP address to the env variable

File: JenkinsFile [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    }
+                }
+            }
+        }
+
+        stage('Run application') {
+            steps {
+                script {
+                    sshagent(credentials: ['ec2']) {
+                        sh 'ssh -o "StrictHostKeyChecking=no" -l ec2-user 172.31.88.196 "nohup java -jar -Dspring.profiles.active=prod -Dspring.datasource.password=${CREDS_DB_PSW} job-search.jar > output.log &"'
```
Comment (oleksandr-fomenko) [inline JenkinsFile]: the same ->IP, artifactName to the env variables

File: JenkinsFile [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                    }
+                }
+                timeout(3) {
+                    waitUntil {
+                        script {
+                            try {
+                                def response = httpRequest 'http://ec2-18-212-196-212.compute-1.amazonaws.com:8080/'
```
Comment (oleksandr-fomenko) [inline JenkinsFile]: suggesting extracting this code to separate function

File: JenkinsFile (hunk 1 above)
Comment (Awesome5040) [reply JenkinsFile]: @oleksandr-fomenko Thanks, all comments fixed in https://github.com/Awesome5040/job-search/pull/5/commits/7579dfd6d68dfb1aecf991dd2063aa9d43331fed

**Pull request merged**